File: archive1/efficient_solution/E_archive1.py
import time

def SumOfMultiplesUnderN(multiple: int, N: int=999) -> float:
    quotient = N // multiple
    sumOfMultiples = multiple * (0.5 * quotient * (quotient+1))
    return sumOfMultiples

if __name__ == "__main__":
    startTime = time.time_ns()
    finalSum = SumOfMultiplesUnderN(3) + SumOfMultiplesUnderN(5) - SumOfMultiplesUnderN(15)
    stopTime = time.time_ns()
    print(finalSum)
    print("Time taken in nano seconds: " + str(stopTime - startTime))

File: archive1/efficient_solution/E_archive1.java
public class E_archive1 {

	public static void main(String[] args) {

		long startTime = System.nanoTime();
		double sum = sumOfdivisibility(3) + sumOfdivisibility(5) - sumOfdivisibility(15);
		System.out.println(sum);
		long stopTime = System.nanoTime();
		System.out.println("Time Taken in nanoseconds: " + (stopTime - startTime));

	}

	public static double sumOfdivisibility(int divisor) {

		int dividend = 999;
		double quotient = dividend / divisor;
		double sum = divisor * 0.5 * quotient * (quotient + 1);
		return sum;
	}
}

